shared/studio/tabs/dataview/state/edits.ts

```diff
@@ -578,7 +581,9 @@ function generateLinkUpdate(
         .filter((change) => change.kind === UpdateLinkChangeKind.Remove)
         .map(({id}) => `'${id}'`)
         .join(", ")}})`,
-      `(select ${linkEdits.escapedLinkTypeName} filter .id in <uuid>{${changes
+      `(select ${
+        linkEdits.linkTypeName === parentObjectTypeName ? "detached " : ""
```

Comment:
Why not always do `detached`?  This doesn't seem sufficient if the loop is not directly with parent.

Reply:
There's not really any reason, I just decided to only add `detached` when it's needed. But I'm not sure what you mean by the 'loop is not directly with parent', the generated update/inserts are only one level deep, with nested inserts flattened into the `with` block.